### ISSUE TITLE
Unable to configure frontend nginx.conf

### ISSUE THREAD

**Issue by mac-chaffee** [kubecost/cost-analyzer-helm-chart#1287]
**Describe the bug**

I run kubecost in a big 96-core server, but the default configuration of the frontend nginx container is to set `worker_processes: auto`, which will launch ~96 processes at startup. This uses a lot of unnecessary RAM and CPU.

Looks like the nginx container does have a way to override that behavior by just setting `NGINX_ENTRYPOINT_WORKER_PROCESSES_AUTOTUNE=1`, but that doesn't work for two reasons:

1. First, we'd need the helm chart to support setting `kubecostFrontend.extraEnv`. I could provide that pull request.
2. Second and more importantly, the file permissions on /etc/nginx/nginx.conf block edits. The container runs as user 1001 but the nginx.conf file is owned by root. That means the scripts in `/docker-entrypoint.d/` which edit nginx.conf fail at startup:

```
/docker-entrypoint.sh: /docker-entrypoint.d/ is not empty, will attempt to perform configuration
/docker-entrypoint.sh: Looking for shell scripts in /docker-entrypoint.d/
/docker-entrypoint.sh: Launching /docker-entrypoint.d/10-listen-on-ipv6-by-default.sh
10-listen-on-ipv6-by-default.sh: info: can not modify /etc/nginx/conf.d/default.conf (read-only file system?)
/docker-entrypoint.sh: Launching /docker-entrypoint.d/20-envsubst-on-templates.sh
/docker-entrypoint.sh: Launching /docker-entrypoint.d/30-tune-worker-processes.sh
30-tune-worker-processes.sh: error: can not modify /etc/nginx/nginx.conf (read-only file system?)
/docker-entrypoint.sh: Configuration complete; ready for start up
```

**To Reproduce**
1. Run `helm install <kubecost> --set kubecostFrontend.resources.limits.cpu=250m` on a multi-core server
2. Observe that even though the container is limited to 1/4 of a CPU, nginx still logs `[notice] 1#1: start worker process N` once per core of the whole server, and that setting `NGINX_ENTRYPOINT_WORKER_PROCESSES_AUTOTUNE=1` fails to fix it

**Expected behavior**
nginx.conf should be configurable via environment variables. It might even be a good idea to set `NGINX_ENTRYPOINT_WORKER_PROCESSES_AUTOTUNE=1` by default since I can't see a downside, but at least providing the option would be nice.

**Collect logs (please complete the following information):**

```
/docker-entrypoint.sh: /docker-entrypoint.d/ is not empty, will attempt to perform configuration
/docker-entrypoint.sh: Looking for shell scripts in /docker-entrypoint.d/
/docker-entrypoint.sh: Launching /docker-entrypoint.d/10-listen-on-ipv6-by-default.sh
10-listen-on-ipv6-by-default.sh: info: can not modify /etc/nginx/conf.d/default.conf (read-only file system?)
/docker-entrypoint.sh: Launching /docker-entrypoint.d/20-envsubst-on-templates.sh
/docker-entrypoint.sh: Launching /docker-entrypoint.d/30-tune-worker-processes.sh
30-tune-worker-processes.sh: error: can not modify /etc/nginx/nginx.conf (read-only file system?)
/docker-entrypoint.sh: Configuration complete; ready for start up
2022/03/02 17:08:09 [warn] 1#1: the "user" directive makes sense only if the master process runs with super-user privileges, ignored in /etc/nginx/nginx.conf:2
nginx: [warn] the "user" directive makes sense only if the master process runs with super-user privileges, ignored in /etc/nginx/nginx.conf:2
2022/03/02 17:08:09 [notice] 1#1: using the "epoll" event method
2022/03/02 17:08:09 [notice] 1#1: nginx/1.21.6
2022/03/02 17:08:09 [notice] 1#1: built by gcc 10.3.1 20211027 (Alpine 10.3.1_git20211027)
2022/03/02 17:08:09 [notice] 1#1: OS: Linux 4.18.0-348.12.2.el8_5.x86_64
2022/03/02 17:08:09 [notice] 1#1: getrlimit(RLIMIT_NOFILE): 1048576:1048576
2022/03/02 17:08:09 [notice] 1#1: start worker processes
2022/03/02 17:08:09 [notice] 1#1: start worker process 24
2022/03/02 17:08:09 [notice] 1#1: start worker process 25
2022/03/02 17:08:09 [notice] 1#1: start worker process 26
2022/03/02 17:08:09 [notice] 1#1: start worker process 27
2022/03/02 17:08:09 [notice] 1#1: start worker process 28
2022/03/02 17:08:09 [notice] 1#1: start worker process 29
2022/03/02 17:08:09 [notice] 1#1: start worker process 30
2022/03/02 17:08:09 [notice] 1#1: start worker process 31
2022/03/02 17:08:09 [notice] 1#1: start worker process 32
2022/03/02 17:08:09 [notice] 1#1: start worker process 33
2022/03/02 17:08:09 [notice] 1#1: start worker process 34
...
2022/03/02 17:08:09 [notice] 1#1: start worker process 113
```

gz#1454

**Comment by keithhand** [comment]
@mac-chaffee, Thank you for this detailed report. We'll likely want to discuss defaulting the `NGINX_ENTRYPOINT_WORKER_PROCESSES_AUTOTUNE` variable, as I'm not sure if there are any consequences on the application side from that change. Let us look over this issue and the PR you pushed to find out the next steps.

**Comment by mac-chaffee** [comment]
#1288  only fixes half the issue: the ability to pass env vars. But the env var doesn't fix this issue due to file permissions, so this issue still exists.

**Comment by AjayTripathy** [comment]
Yeah, I see the issue. It sounds like we'll have to use our own build of nginx with a separate dockerfile since it's a nonstarter for many users to allow any containers to run as root. 
https://stackoverflow.com/questions/47064845/unable-to-change-nginx-conf-file

**Comment by dwbrown2** [comment]
@AjayTripathy proposed next steps? Looks like you have a PR outstanding?

**Comment by AjayTripathy** [comment]
I have a fix I need to test in our frontend container build repo that changes the ownership of the config file in the docker script. Not sure if that plus the env variable will be sufficient, but plan to test soon.

**Comment by mac-chaffee** [comment]
Oh @AjayTripathy check out https://github.com/nginxinc/docker-nginx-unprivileged as a base image instead of `nginx:latest`. It supports that `NGINX_ENTRYPOINT_WORKER_PROCESSES_AUTOTUNE` variable, but [should have the file permissions issue fixed](https://github.com/nginxinc/docker-nginx-unprivileged/blob/5481a8967a0064a8948c90215e31f24012bd1e67/Dockerfile-debian.template#L106).

**Comment by AjayTripathy** [comment]
Great find @mac-chaffee ! Let me investigate that as well if my quick chmod on the image doesn't work.

**Comment by AjayTripathy** [comment]
On my list for this week!

**Comment by AjayTripathy** [comment]
@mac-chaffee thanks for the tip, I believe this is working. Can you try out 

`gcr.io/kubecost1/frontend:nginx-unprivileged` to see if this is working for you? Here are my logs, not that 30-tune-worker-processes.sh runs.

```
atripathy@Ajays-MacBook-Pro builds % kubectl logs kubecost-cost-analyzer-695c7b6bcc-8bqts -n kubecost -c cost-analyzer-frontend
/docker-entrypoint.sh: /docker-entrypoint.d/ is not empty, will attempt to perform configuration
/docker-entrypoint.sh: Looking for shell scripts in /docker-entrypoint.d/
/docker-entrypoint.sh: Launching /docker-entrypoint.d/10-listen-on-ipv6-by-default.sh
10-listen-on-ipv6-by-default.sh: info: can not modify /etc/nginx/conf.d/default.conf (read-only file system?)
/docker-entrypoint.sh: Launching /docker-entrypoint.d/20-envsubst-on-templates.sh
/docker-entrypoint.sh: Launching /docker-entrypoint.d/30-tune-worker-processes.sh
/docker-entrypoint.sh: Configuration complete; ready for start up
2022/03/29 04:41:32 [notice] 1#1: using the "epoll" event method
2022/03/29 04:41:32 [notice] 1#1: nginx/1.20.2
2022/03/29 04:41:32 [notice] 1#1: built by gcc 10.3.1 20210424 (Alpine 10.3.1_git20210424) 
2022/03/29 04:41:32 [notice] 1#1: OS: Linux 5.4.170+
2022/03/29 04:41:32 [notice] 1#1: getrlimit(RLIMIT_NOFILE): 1048576:1048576
```

**Comment by mac-chaffee** [comment]
@AjayTripathy Looks good!

```
/docker-entrypoint.sh: /docker-entrypoint.d/ is not empty, will attempt to perform configuration
/docker-entrypoint.sh: Looking for shell scripts in /docker-entrypoint.d/
/docker-entrypoint.sh: Launching /docker-entrypoint.d/10-listen-on-ipv6-by-default.sh
10-listen-on-ipv6-by-default.sh: info: can not modify /etc/nginx/conf.d/default.conf (read-only file system?)
/docker-entrypoint.sh: Launching /docker-entrypoint.d/20-envsubst-on-templates.sh
/docker-entrypoint.sh: Launching /docker-entrypoint.d/30-tune-worker-processes.sh
/docker-entrypoint.sh: Configuration complete; ready for start up
2022/03/29 15:39:54 [notice] 1#1: using the "epoll" event method
2022/03/29 15:39:54 [notice] 1#1: nginx/1.20.2
2022/03/29 15:39:54 [notice] 1#1: built by gcc 10.3.1 20210424 (Alpine 10.3.1_git20210424)
2022/03/29 15:39:54 [notice] 1#1: OS: Linux 4.18.0-348.12.2.el8_5.x86_64
2022/03/29 15:39:54 [notice] 1#1: getrlimit(RLIMIT_NOFILE): 1048576:1048576
2022/03/29 15:39:54 [notice] 1#1: start worker processes
2022/03/29 15:39:54 [notice] 1#1: start worker process 11
```

**Comment by kbrwn** [comment]
Hello @mac-chaffee thanks for confirming. The fix for this has shipped in the [Kubecost v1.92.0 release](https://github.com/kubecost/cost-analyzer-helm-chart/releases/tag/v1.92.0).
